src/layman/layer/rest_workspace_layer.py

```diff
@@ -250,6 +253,7 @@ def patch(workspace, layername):
     else:
         delete_from = 'layman.layer.micka.soap'
 
+    layer.set_geodata_type(geodata_type)
     util.patch_layer(
         layer,
```

Comment:
Let's discuss
- How to name `layer` parameter in `patch_layer` methods. It can be e.g. `old_layer`, `new_layer`. Also, I would like to be consistent in setting either all new attribute values, or no new attribute values. Otherwise it's quite hard to realize what attribute has new value and what attribute has old value.
- If and how to enable creating Layer instance with specific values that are not yet presented in prime DB schema.

(related also to maps)

Ideas
- we can use something like `LayerMock` instance for `new_layer`.
- we can (1) enable passing values to `Layer` constructor without reading them from prime DB schema, (2) implement custom `.replace(**new_kwargs)` and (3) use something like `new_layer = old_layer.replace(geodata_type='vector', style_type='qml')`